### DISCUSSION TITLE
github auto merge process blocked from protected branch with code review required

### DISCUSSION THREAD

**Discussion by ns-ggeorgiev** [renovatebot/renovate#15750] [Request Help]
Hi,

I am trying to get auto merging to works with renovate when reviewers are required.

What I am experiencing right now is, renovate generates the PR, and even if all checks pass, if the branch is configured to need a code review approved before merge, the PR just lingers.

Is there a solution for this today?

Obviously, removing the requirement is not a solution or even workaround - it is not an option.

Thanks

**Comment by rarkins** [comment]
1. Install the renovate approve bot app, or
2. Allow renovate to bypass branch protections

**Reply by ns-ggeorgiev** [reply]
I am self hosting - is allow the user that I am using going to solve the issue?

Note: I do not want direct merge, I still want all the checks to pass before the merge is made, I just want to eliminate the step of code review approval.

**Reply by ns-ggeorgiev** [reply]
oh, nevermind, it seems the bot is what I need

**Reply by ns-ggeorgiev** [reply]
@rarkins: I need the owners scenario covered too. What I am thinking is just creating dedicated github account for that goal that teams will add as owners in their list. Then I need something to track for the PRs and approve on behalf of that user.

I wonder if you are willing to enhance the current approval bot to accept a token to impersonate the approval, instead of doing it as the app itself.

Also wouldn't this be better done from the renovate itself as part of the creation of the PRs, instead creating something else to chaice the PRs after that?

**Reply by rarkins** [reply]
Apps can't be codeowners, so you would need to use a PAT anyway for that scenario.

Apps have the convenience of anyone can install, and can install into every repo in one go, so the solution has been very helpful to most people with this requirement.

If you want to do it yourself with a PAT, I recommend you just do it in a GitHub Action.

**Reply by rarkins** [reply]
Apps can't approve themselves either, which is why it needs to be a separate bot. In fact we have two approve bots for people who require two approvers.

**Reply by ns-ggeorgiev** [reply]
@rarkins github action make sense - thanks
